### PR TITLE
Add a 'stubtest' flag to METADATA.toml

### PR DESCRIPTION
**Pull request by srittau**
Remove the EXCLUDE_LIST from the stubtest_third_party script.

### CODE REVIEW

**Comment by srittau** [comment]
This has the advantage of keeping information about a package in one place. It also means that stubtest is run when setting the flag to `true` (or removing it).

**Comment by Akuli** [comment]
I'm fine with merging this, but I still don't really like it or see what problem it solves. Excluding anything else than what's already excluded should be discouraged, not documented in CONTRIBUTING.md.

A hard-coded list in a `.py` file looks like it should be eventually gone, and indeed it should, but IMO that should be done by fixing the stubs, not by moving the configuration elsewhere.

**Comment by srittau** [comment]
* There will always be cases like this.
* All information about a distribution belongs in one place for easy discoverability. It shouldn't be scattered around the distribution directory and potentially multiple scripts.
* "Hiding" this in some script does not help. For example, I wasn't aware/had forgotten that we skip some tests. If I were to work on, e.g. pycurl, I'm much more likely to discover this entry in `METADATA.toml` than tugged away in some test script. That's more likely to be an issue for external contributors. I also don't want to have to modify some support scripts when working on a certain distribution.
* Changing `METADATA.toml` inside a distribution directory will trigger distribution-specific tests. (Currently this affects only stubtest, but we might need to run certain tests only when a distribution changes as typeshed grows.) Changing scripts will not do this.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -177,6 +177,8 @@ supported:
   [removing obsolete third-party libraries](#third-party-library-removal-policy).
   It contains the first version of the corresponding library that ships
   its own `py.typed` file.
+* `stubtest` (default: `true`): Whether stubtest should be run against this
+  package. Please add a comment with the reason when disabling stubtest.
```
Comment (Akuli) [inline CONTRIBUTING.md:181]:
```suggestion
  package. Please avoid setting this to `false`, and add a comment if you have to.
```

File: stubs/six/METADATA.toml [hunk 2]
```diff
@@ -1,2 +1,3 @@
 version = "1.16.*"
 python2 = true
+stubtest = false
```
Comment (Akuli) [inline stubs/six/METADATA.toml:3]:
```suggestion
stubtest = false  # TODO: figure out why this is needed
```

**Comment by srittau** [comment]
I plan to review a few of those after this gets merged.

**Review by Akuli — APPROVED** [review]

**Pull request merged**